### PR TITLE
[REVIEW] implement .min() .max() for DatetimeColumn

### LINKED ISSUE #953: [FEA] Implement .min() and .max() of a column
**Is your feature request related to a problem? Please describe.**
I need to calculate the min and max of a column with datatype numpy.datetime64.

I have tried _df['date'].min()_ and _df['date'].max()_, but they seem to do not be implemented yet.

If I do _min(df['date'])_ or _max(df['date'])_, it takes about 6 seconds each, compared to miliseconds within Pandas.

**Describe the solution you'd like**
.min() and .max() functions implemented.

**Describe alternatives you've considered**
_min(df['date'])_ 
 _max(df['date'])_

**Additional context**
I am porting code from pandas to cudf

### PR DESCRIPTION
**Pull request by kovaltan**
Implement .min() .max() for DatetimeColumn while `NumericalColumn` has those funcitons.

Close #953 ([FEA] Implement .min() and .max() of a column)

- [x] Implement .min() .max() for DatetimeColumn
- [x] add python test case for DatetimeColumn.min(), .max()
- [x] return datetime64[ms] type for GDF_DATE64

### CODE REVIEW

File: python/cudf/bindings/cudf_cpp.pyx [hunk 1]
```diff
@@ -72,16 +72,20 @@ cdef gdf_dtype get_dtype(dtype):
...
         GDF_INT64:   scalar.data.si64,
         GDF_INT32:   scalar.data.si32,
         GDF_INT16:   scalar.data.si16,
         GDF_INT8:    scalar.data.si08,
-        GDF_DATE32:  scalar.data.dt32,
-        GDF_DATE64:  scalar.data.dt64,
-        GDF_TIMESTAMP: scalar.data.tmst,
+        GDF_DATE32:  np.array(scalar.data.dt32).astype('datetime64[D]'),
+        GDF_DATE64:  np.array(scalar.data.dt64).astype('datetime64[ms]'),
+        GDF_TIMESTAMP: np.array(scalar.data.tmst).astype('datetime64[ns]'),
```
Comment (kkraus14) [inline python/cudf/bindings/cudf_cpp.pyx]: Is `GDF_TIMESTAMP` always nanosecond resolution in libcudf? My understanding is that the resolution can be arbitrary and is an attribute of the `gdf_column` object.

Comment (jrhemstad) [reply python/cudf/bindings/cudf_cpp.pyx]: Yeah, the resolution is technically a separate attribute of the `gdf_column`. Nothing in `libcudf` actually uses or respects that resolution right now :)

Comment (kkraus14) [reply python/cudf/bindings/cudf_cpp.pyx]: Thanks @jrhemstad, will just merge this for now and can be resolved when we actually begin using `GDF_TIMESTAMP`

**Comment by kovaltan** [comment]
> Changes look good. One question about `GDF_TIMESTAMP`, but it's not being used in the Python yet anyway.

Thanks for merging, it seems we need `gdf_dtype_extra_info`  in `gdf_scalar` to treat the resolution of `GDF_TIMESTAMP`
I've file a issue #1449

**Pull request merged**